editoast/src/views/layers/geo_json.rs

```diff
@@ -0,0 +1,50 @@
+use mvt::GeomType;
+
+use rocket::serde::json::Value as JsonValue;
+use serde::{Deserialize, Deserializer};
+
+/// GeoJson used by the application
+pub enum GeoJsonType {
+    Point,
+    MultiPoint,
+    LineString,
+    MultiLineString,
+}
+
+impl GeoJsonType {
+    /// Gets MVT geom type corresponding to the GeoJson type
+    pub fn as_mvt_geom_type(&self) -> GeomType {
+        match self {
+            GeoJsonType::Point => GeomType::Point,
+            GeoJsonType::MultiPoint => GeomType::Point,
+            GeoJsonType::LineString => GeomType::Linestring,
+            GeoJsonType::MultiLineString => GeomType::Linestring,
+        }
+    }
+}
+
+impl<'de> Deserialize<'de> for GeoJsonType {
+    fn deserialize<D>(deserializer: D) -> Result<Self, D::Error>
+    where
+        D: Deserializer<'de>,
+    {
+        let s = String::deserialize(deserializer)?;
+        Ok(match s.as_str() {
+            "Point" => GeoJsonType::Point,
+            "MultiPoint" => GeoJsonType::MultiPoint,
+            "LineString" => GeoJsonType::LineString,
+            "MultiLineString" => GeoJsonType::MultiLineString,
+            value => panic!("Unsupported MVT geom_type {value}"),
+        })
+    }
+}
```

Comment:
You don't need to do this you can use serde derive: https://serde.rs/derive.html